### PR TITLE
Extend construct method for constructor invocations on resources

### PR DESCRIPTION
**Pull request by 1nf0rmagician**
Before, we created new resources using reflection and returned them always unsafed. This does not allow resources to act on the resource graph within their constructor methods, however. Hence, we added a conditional invocation to the resource modification interface, which creates a new resource in the resource graph with all required initializations. The behaviour of the endpoint is thereby extended with the option to create such resources with their respective constructors.

### CODE REVIEW

File: src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs [hunk 1]
```diff
@@ -144,30 +145,48 @@ public ActionResult<Entry> InvokeMethod(long id, string method, Entry parameters
...
+        private ActionResult<ResourceModel> Construct(string type, MethodEntry method)
+        {
+            try
+            {
+                var id = _resourceManagement.Create(_resourceTypeTree[type].ResourceType, r => EntryConvert.InvokeMethod(r, method, _serialization));
+                return GetDetails(id);
+            }
+            catch (Exception e)
+            {
+                if (e is ArgumentException or SerializationException or ValidationException)
```
Comment (seveneleven) [inline src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs:184]: Would it be possible to validate the input at the beginning?

File: src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs [hunk 2]
```diff
@@ -187,11 +206,11 @@ public ActionResult<ResourceModel> Save(ResourceModel model)
                     resourcesToSave.Skip(1).ForEach(id => _resourceManagement.Modify(id, r => true));
                 });
 
-                return Ok(GetDetails(id));
+                return GetDetails(id);
```
Comment (seveneleven) [inline src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs:209]: Why did you remove the explicit `Ok` here? I like the readability aspect of it.

File: src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs (hunk 1 above)
Comment (1nf0rmagician) [reply src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs:184]: Unfortunately that is not so easy. The validation logic is nested in the resource management, we would either need to pull that out of it and only use it there and here, or provide a method on the facade. But that is not part of this PR, I just added the catch block here because I stumbled upon the issue by accident. This way you at least get the correct response code 🙈

File: src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs (hunk 2 above)
Comment (1nf0rmagician) [reply src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs:209]: Yeah that is why I added it in a recent commit, problem is GetDetails is already a controller action that returns Ok(resourceModel), i.e. wrapping it in another Ok() call causes the response object to look like this
```json
{
    result: null,
    value: // the actual resource model that should be serialized
}
```
This is not detected by the ui compiler or swagger, but causes errors when saving resources for example.

**Review by seveneleven — APPROVED** [review]
